src/CommitmentsV2/SFA.DAS.CommitmentsV2.MessageHandlers/NServiceBus/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Data.Common;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using NServiceBus;
+using SFA.DAS.CommitmentsV2.MessageHandlers.Configuration;
+using SFA.DAS.NServiceBus;
+using SFA.DAS.NServiceBus.NewtonsoftJsonSerializer;
+using SFA.DAS.NServiceBus.NLog;
+using SFA.DAS.NServiceBus.SqlServer;
+using SFA.DAS.NServiceBus.StructureMap;
+using SFA.DAS.UnitOfWork.NServiceBus;
+using StructureMap;
+
+namespace SFA.DAS.CommitmentsV2.MessageHandlers.NServiceBus
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection ConfigureNServiceBus(this IServiceCollection services)
+        {
+            return services
+                .AddSingleton<IEndpointInstance>(s =>
+                {
+                    var container = s.GetService<IContainer>();
+                    var hostingEnvironment = s.GetService<IHostingEnvironment>();
+
+                    // This will need to be resolved or injected
+                    var configurationSection = new CommitmentsV2Configuration();
+                    configurationSection.ServiceBusConnectionString =
+                        "Endpoint=sb://das-at-shared-ns.servicebus.windows.net/;SharedAccessKeyName=ReadWrite;SharedAccessKey=8NRQlHb+mE+/2DiRUVpeFThHD5WUYpO4dwCzue4aNQk=";
```

Comment:
Should this be in code? Do we want the AT shared access key in code?

Reply:
removed